src/files/FileLocationEntity.js

```diff
@@ -0,0 +1,18 @@
+import { Entity } from '../es6/Entity';
+
+/**
+ * FileLocationEntity class representation of a d2l content file entity.
+ */
+export class FileLocationEntity extends Entity {

```

Comment:
Open to any other name suggestions! The `ContentHtmlFileEntity` has a `getFilesHref()` which, if the link is called, would return this entity. This entity has the link which returns the data of the actual file.
![html-sdk](https://user-images.githubusercontent.com/52549862/113538432-d40f2380-95a0-11eb-8034-45219d27910f.gif)
I've been trying to work out how it will look on the activities side, but I believe that once `ContentHtmlFileEntity` is initialized, we would then need to fetch this entity, get the `getFileLocationHref()` then fetch the actual file data.

Reply:
Could it just be a `FileEntity`? It's essentially the siren entity representing the file, which you then use to get the href to the file resource itself, but it could (at least theoretically) have other file metadata on it beyond just the location I think?